### PR TITLE
Updated the docker-compose.yml to include redis for ratelimiting

### PR DESCRIPTION
**Pull request by rgautam98**
This docker-compose includes the image from bitnami. We are using the latest revision. The image also supports master nodes, replication and cluster configuration. I however ignored all those right now as I believe ops team will have a better understanding of the deployment. 

Having said that, I belive in our production, it would be great to have an elastic cache cluster so that high availability and clustering can be given to a third party provider. 

If for any reason we dont have the choice, the bitnami image will do all of that for us.

### CODE REVIEW

**Review by s0 — CHANGES_REQUESTED** [review]
Good start, but I'd like to see some usage to make sure that it's all set up correctly. In this case, just a simple endpoint that checks that it can connect to redis okay, and can store and retreive some cached value.

@astroash is working on lots of changes atm, so it may not be a great time to implement this. I'll reassign to her to let us know when it makes sense to add these changes, at which point she can open it up to you to implement again.

Beyond this, using the image that serban indicated will be needed too.

File: docker-compose.yml [hunk 1]
```diff
@@ -36,5 +37,19 @@ services:
       - 4000:5100
     depends_on:
       - db
+
+  redis:
+    image: bitnami/redis:6.2
```
Comment (teodorescuserban) [inline docker-compose.yml:42]:
```suggestion
    image: unocha/redis:6
```

File: docker-compose.yml [hunk 2]
```diff
@@ -36,5 +37,19 @@ services:
...
+    restart: always
+    environment:
+      # - ALLOW_EMPTY_PASSWORD=yes # Basically No auth. For development only
+      - REDIS_PASSWORD=password123 # If you login via the tty shell, use the command redis-cli -a $REDIS_PASSWORD to access the cli
```
Comment (teodorescuserban) [inline docker-compose.yml:46]:
```suggestion
```

File: docker-compose.yml [hunk 3]
```diff
@@ -36,5 +37,19 @@ services:
...
+    ports:
+      - 6379:6379
+    volumes:
+      - redis_data:/bitnami/redis/data
```
Comment (teodorescuserban) [inline docker-compose.yml:50]:
```suggestion
      - redis_data:/var/lib/redis
```

File: docker-compose.yml [hunk 4]
```diff
@@ -1,4 +1,5 @@
-version: "3.8"
+
+version: "3.3"
 services:
   db:
     image: postgres:12.4-alpine
```
Comment (teodorescuserban) [inline docker-compose.yml:5]:
```suggestion
    image: unocha/postgres:12
```

**Review by teodorescuserban — CHANGES_REQUESTED** [review]
Postgres env var names become:
- `POSTGRES_PASSWORD` -> `PGSQL_PASS`
- `POSTGRES_USER` -> `PGSQL_USER`
- `POSTGRES_DB` -> `PGSQL_DB`

**Comment by astroash** [comment]
@rgautam98 new docker configuration has been merged in https://github.com/UN-OCHA/hpc-api/pull/14 . Let me know if you want to go through any of the changes together on call :)

@teodorescuserban thanks for the pointers for getting this PR moved over to the unocha docker image

**Comment by Pl217** [comment]
This PR gained attention from various people, but author hasn't responded to any in quite a while, it's been almost 9 months since opening the PR.

If we also consider that this PR just registers Redis and doesn't do anything with it, there is no value in keeping this dead PR open, so I'm going to close it.

**Pull request closed without merging**